Heroku/config/detekt/detekt.yml

```diff
@@ -53,7 +53,25 @@ complexity:
 
   NamedArguments:
     # Exclude in test code, mocking doesn't need to have explicit named arguments, since a minor mistake will fail the tests.
-    excludes: [ '**/test/**', '**/androidTest/**', '**/commonTest/**', '**/jvmTest/**', '**/jsTest/**', '**/iosTest/**' ]
+    excludes: [ '**/unitTest/**', '**/functionalTest/**', '**/integrationTest/**', '**/integrationExternalTest/**' ]
+
+  StringLiteralDuplication:
+    excludes: [ '**/unitTest/**', '**/functionalTest/**', '**/integrationTest/**', '**/integrationExternalTest/**' ]
+
+  TooManyFunctions:
+    excludes: [ '**/unitTest/**', '**/functionalTest/**', '**/integrationTest/**', '**/integrationExternalTest/**' ]
+
+
+exceptions:
+  
+  InstanceOfCheckForException:
+    excludes: [ '**/unitTest/**', '**/functionalTest/**', '**/integrationTest/**', '**/integrationExternalTest/**' ]
+
+  ThrowingExceptionsWithoutMessageOrCause:
+    excludes: [ '**/unitTest/**', '**/functionalTest/**', '**/integrationTest/**', '**/integrationExternalTest/**' ]
+
+  TooGenericExceptionCaught:
+    excludes: [ '**/unitTest/**', '**/functionalTest/**', '**/integrationTest/**', '**/integrationExternalTest/**' ]
```

Comment:
@BraisGabin re https://github.com/detekt/detekt/pull/5797, this doesn't stand. After migrating to Gradle Test Suites, I just spent 20 minutes whack-a-moling which rules need to be excluded for tests (because the folder names changed from `src/test/kotlin`. Also noticed that my `:test-helpers` module was treated as test code, even though it's a library of production code made for testing which has its own tests.

Ideally this would be as simple as:
```
config:
  exclude-aliases:
    test: [ '**/unitTest/**', '**/functionalTest/**', '**/integrationTest/**', '**/integrationExternalTest/**' ]

# and the rest of the config without any repetition of defaults or folder patterns.
```
internally we could define a few aliases:
```
config:
  exclude-aliases:
    jvm-test: ...
    kmp-test: ...
    android-test: ...
    test: [ ':jvm-test:', ':kmp-test:', ':android-test:' ]
```
and then use these aliases to configure default excludes.
`` ':alias-name:' `` is just an example, syntax up for debate.

Reply:
I completely agree that our configuration is far away from perfect. I really think that the best way to work with our current configuration is to have it all written down in your own configuration file. And then use other tools to keep it aligned with the one from detekt.

We could even create a smart merger. We know the semantics of our configuration file so it should create less conflicts than pure file merging.

But I must say that I would love to hear ideas to make our configuration more maintenable for our users.